### ISSUE TITLE
Add support for automatic JSON conversion

### ISSUE THREAD

**Issue by njh** [njh/ruby-mqtt#60]

**Comment by njh** [comment]
Also (automatic?) gzip compression?

    There is a magic number at the beginning of the file. Just read the first two bytes and check if they are equal to 0x1f8b

**Comment by njh** [comment]
Yes, that is partly why I didn't rush into implementing it. But it could be that there is a way 

However JSON is very popular with MQTT:

1. MQTT recommends (but doesn't require) using human readable, text based payloads
2. JSON is very concise/compact
3. JSON maps directly into high-level programming language features.

I have been very frustrated with HTTP libraries, where there first thing I have to do in my code is ```if content_type = "application/json" then JSON.parse```.

There would certainly need to be a way of turning it on and off. I was imagining something like:

    client.convert_payload = :auto    # Automatically parse payloads, hashes/arrays become JSON
    client.convert_payload = false     # Disable payload conversion
    client.convert_payload = :json     # Parse payloads as JSON, generate JSON

**Comment by skandragon** [comment]
Perhaps as an option, but if all I want is MQTT, I should just get MQTT.  Handling JSON from within Ruby is stupid simple, and I'd keep your code base clean and focused.

Besides, what happens when the next person asks for XML?  :)
